packages/starlight/__tests__/test-config.ts

```diff
@@ -4,11 +4,19 @@ import { getViteConfig } from 'astro/config';
 import type { z } from 'astro/zod';
 import { vitePluginStarlightUserConfig } from '../integrations/virtual-user-config';
 import { StarlightConfigSchema } from '../utils/user-config';
+import type { AstroConfig } from 'astro';
+
+export function defineVitestConfig(
+	config: z.input<typeof StarlightConfigSchema>,
+	opts?: Partial<AstroConfig>
+) {
+	const root = opts?.root ?? new URL('./', import.meta.url);
+	const srcDir = opts?.srcDir ?? new URL('./src/', root);
+	const build = opts?.build ?? ({ format: 'directory' } as AstroConfig['build']);
```

Comment:
Should allow us to set this without the assertion I think?

```suggestion
	const build = opts?.build ?? { format: 'directory' };
```